src/main/java/org/dependencytrack/tasks/maintenance/ProjectMaintenanceTask.java

```diff
@@ -0,0 +1,114 @@
+import static org.dependencytrack.util.LockProvider.executeWithLock;
+import static org.dependencytrack.util.TaskUtil.getLockConfigForTask;
+
+public class ProjectMaintenanceTask implements Subscriber {
+
+    private static final Logger LOGGER = Logger.getLogger(ProjectMaintenanceTask.class);
+
+    @Override
+    public void inform(final Event event) {
+        if (!(event instanceof ProjectMaintenanceEvent)) {
+            return;
+        }
+
+        final long startTimeNs = System.nanoTime();
+        try (final Handle jdbiHandle = openJdbiHandle()) {
+            LOGGER.info("Starting project maintenance");
+            final Statistics statistics = executeWithLock(
+                    getLockConfigForTask(ProjectMaintenanceTask.class),
+                    () -> informLocked(jdbiHandle));
+            if (statistics == null) {
+                LOGGER.info("Task is locked by another instance; Skipping");
+                return;
+            }
+
+            final var taskDuration = Duration.ofNanos(System.nanoTime() - startTimeNs);
+            LOGGER.info("Completed in %s: %s".formatted(taskDuration, statistics));
+        } catch (Throwable e) {
+            final var taskDuration = Duration.ofNanos(System.nanoTime() - startTimeNs);
+            LOGGER.error("Failed to complete after %s".formatted(taskDuration), e);
+        }
+    }
+
+    private record Statistics(int deletedInactiveProjects) {
+    }
+
+    private Statistics informLocked(final Handle jdbiHandle) {
+        assertLocked();
+
+        final var configPropertyDao = jdbiHandle.attach(ConfigPropertyDao.class);
+
+        var retentionType = configPropertyDao.getValue(MAINTENANCE_INACTIVE_PROJECTS_RETENTION_TYPE, String.class);
+
+        int batchSize = 100;
+        AtomicInteger numDeletedTotal = new AtomicInteger(0);
+
+        if (retentionType.equals("AGE")) {
+            int retentionDays = configPropertyDao.getValue(MAINTENANCE_INACTIVE_PROJECTS_RETENTION_DAYS, Integer.class);
+            final Duration retentionDuration = Duration.ofDays(retentionDays);
+            Integer numDeletedLastBatch = null;
+            while (numDeletedLastBatch == null || numDeletedLastBatch > 0) {
+                numDeletedLastBatch = withJdbiHandle(
+                        batchHandle -> {
+                            final var projectDao = jdbiHandle.attach(ProjectDao.class);
+                            return projectDao.deleteInactiveProjectsForRetentionDuration(retentionDuration, batchSize);
+                        });
+                numDeletedTotal.addAndGet(numDeletedLastBatch);
+            }
+        } else {
+            int retentionCadence = configPropertyDao.getValue(MAINTENANCE_INACTIVE_PROJECTS_RETENTION_CADENCE, Integer.class);
+            Integer projectLastBatch = null;
+            while (projectLastBatch == null || projectLastBatch > 0) {
+                projectLastBatch = withJdbiHandle(
```

Comment:
Since you're doing more than one query here, better do it in a transaction:

```suggestion
                projectLastBatch = inJdbiTransaction(
```